packages/next/pages/_document.tsx

```diff
@@ -158,6 +166,10 @@ export class Head extends Component<
     const { _devOnlyInvalidateCacheQueryString } = this.context
 
     return dedupe(dynamicImports).map((bundle: any) => {
+      // Only render .js files here
+      if (!bundle.file.endsWith(formatModernPath('.js'))) {
```

Comment:
This behavior makes it preload *only modern chunks* if the feature is turned on.

Ditto for the others.

Is that what we want?
I guess it makes sense since `<link rel="preload" />` is a newer browser feature &mdash; I'm just double checking.

Maybe we want to user-agent sniff this?

Reply:
We decided to preload only modern when the feature is enabled for now. User agent sniffing is something we want to look into, but right now that means the server responses can't be cached or should include the UA string as part of the cache?